### ISSUE TITLE
Possible bug with teleop_twist_joy and Logitech F710 game controller

### ISSUE THREAD

**Issue by altineller** [ros2/teleop_twist_joy#37]
I am on ROS humble, and using teleop_twist_joy with Logitech f710 controller for a while. After a recent update with humble, the joystick started acting erroneously.
Upon closer inspection, I have found that the joystick reports x and y axes as 0.99 initially. So when you push the security button, it will send the cmdvel command derived from x=0.99, y=0.99 resulting in robot to move in full speed in a circular pattern, without touching the joystick. So the solution is move the joysticks before touching the security button and it will work normally as before.
But once the joystick is plugged in it reads 0.99 for all axes, and not 0.0 as it was before. I understand this could be on various levels, maybe some change in linux kernel, or the module that reads /dev/input/js0 - or the teleop_twist_joy package itself.

It is not a problem for me but if some other people are using the same setup, after software update, the teleop_twist_joy package will work erroneously, and it might cause a crash of the robot or accident, by accelerating with 0 throttle input, but security button pushed.

Best Regards, Can

**Comment by clalancette** [comment]
There has been a recent update to teleop_twist_joy (2.4.5), but I highly doubt it would have caused your problem.  That was just a change to how the library was linked.

I do kind of remember seeing this kind of thing before.  I'd make sure to reboot the computer, and/or reboot the joystick itself (take out the batteries), and also unplug and replug the USB dongle.  This may help the situation.

**Comment by padhupradheep** [comment]
We found this [problem](https://github.com/neobotix/neo_teleop2/issues/17#issuecomment-1419222263) 5 months ago. I have explained about it in detail in the link.  I've also added how to work around. 

As mentioned over there. I think the problem has taken it's root in Logitech f710 driver and probably needs to be somehow handled in the `Joy` node. We'd be happy to provide a fix for this issue soon.

P.S.: We do not use `teleop_twist_joy` package but rather our own home brewed package that takes the inputs from the joy command. 

> There has been a recent update to teleop_twist_joy (2.4.5), but I highly doubt it would have caused your problem. That was just a change to how the library was linked.

Definitely not the cause. We can also see the issue in Ubuntu 20.04. That means, the issue can also be reproduced in ROS-Noetic.

**Comment by padhupradheep** [comment]
Could we please move this issue to https://github.com/ros-drivers/joystick_drivers ? Since the issue originates from there. 

Thanks

**Comment by clalancette** [comment]
> Could we please move this issue to https://github.com/ros-drivers/joystick_drivers ? Since the issue originates from there.

Unfortunately, GitHub does not allow me to transfer issues across organizations, so I can't do that.  My suggestion is that you open up another issue over at that repository; once that is done, I'll close this one in favor of that.
